parlai/core/torch_agent.py

```diff
@@ -356,11 +366,48 @@ def observe(self, observation):
         return observation
 
     def act(self):
-        """Calls batch_act with the singleton batch"""
+        """Calls batch_act with the singleton batch."""
         return self.batch_act([self.observation])[0]
 
-    def batch_act(self):
-        raise NotImplementedError("Abstract class: user must implement batch_act")
+    def batch_act(self, observations):
+        batch_size = len(observations)
+        # initialize a table of replies with this agent's id
+        batch_reply = [{'id': self.getID()} for _ in range(batch_size)]
+
+        is_training = any(['labels' in obs for obs in observations])
```

Comment:
We should do high level comments in this method, as I expect users to be falling back to it to help understand what is happening around their implementation.

Reply:
Awesome, thank you!